### PR TITLE
Using urls from opensites.txt

### PR DESCRIPTION
**Pull request by AugustinJose1221**
Add new urls to this text files to increase the number of urls with CC License

### CODE REVIEW

File: app/indexer/detect_open.py [hunk 1]
```diff
@@ -3,11 +3,16 @@
...
-    for i in ["wikipedia.org", "stackoverflow.com"]:
-        if i in url:
-            return True
-    return False
+    with open("opensites.txt",'r') as file: #Using urls specified in opensites.txt
+        '''Checking for wikipedia or SO page'''
+        content = file.readlines()
+        urls = []
+        for i in content:
+            urls.append(i[0:int(len(i)-1)])
```
Comment (asdofindia) [inline app/indexer/detect_open.py]: Would one of the solutions in https://stackoverflow.com/questions/12330522/reading-file-without-newlines-in-python be cleaner?

Comment (minimalparts) [reply app/indexer/detect_open.py]: I agree with @asdofindia . I personally use file.read().splitlines(). But whichever you prefer...

File: app/indexer/detect_open.py [hunk 2]
```diff
@@ -3,11 +3,12 @@
 
 
 def open_site(url):
-    '''Checking for wikipedia or SO page'''
-    for i in ["wikipedia.org", "stackoverflow.com"]:
+    file = open("Opensites.txt",'r').readlines()
```
Comment (minimalparts) [inline app/indexer/detect_open.py]: Should be 'opensites' (not capitalised).

File: app/indexer/controllers.py [hunk 3]
```diff
@@ -38,6 +38,9 @@ def from_file():
         # filename = secure_filename(file.filename)
         file.save(join(dir_path, "urls_to_index.txt"))
         return render_template('indexer/progress_file.html')
+    else:
+        flash(".txt file required","text")
```
Comment (minimalparts) [inline app/indexer/controllers.py]: This is great. Just two small comments:
* the error message is not obvious to the user when it appears. How about making it red? See https://stackoverflow.com/questions/44569040/change-color-of-flask-flash-messages
* when the template reloads, the submit buttons end up going over the edge of the box at the bottom. You could just hack it by making the box a little bigger in the first place... Also, curiously, the number of recorded pages at the top of the page disappears. Not sure why.

**Comment by minimalparts** [comment]
Thanks a lot for the commits! In general, it is better to keep pull requests for different issues separate. For instance, I could now accept the PR for the opensites list (with the small caveat that the file name is wrong -- see my comment). But it would be better to have a separate PR for the message flashing. No big deal, but just to keep in mind for the future. Thanks! :)

**Comment by nandajavarma** [comment]
This PR is quite outdated and hasn't had any traction recently. Closing for now.

**Pull request closed without merging**